miden-tx/tests/common/mod.rs

```diff
@@ -78,3 +87,61 @@ impl DataStore for MockDataStore {
         Ok(self.account.code().module().clone())
     }
 }
+
+// HELPER FUNCTIONS
+// ================================================================================================
+pub fn get_new_key_pair_with_advice_map() -> (KeyPair, ([Felt; 4], Vec<Felt>)) {
```

Comment:
It looks like we some redundancy here. We return the `KeyPair` which contains the `PublicKey` and secret key. We also return `(PublicKey as Word, Vec[PublicKey as Word, SecretKey as Word]))`. So the public key is contained in 3 of the returned components and the secret key in two. You could just return `(Word, Word)`. One word for the public key and one word for the private key.

This could then be invoked like:

`let (pk, sk) = get_new_keypair();`

When you define the entry for the `TransactionScript` inputs it would be constructed as `(pk, [pk, sk].concat())`.

I think this is a bit cleaner.

Reply:
That makes a lot of sense

Reply:
I only don't get this 

```
    let pk_sk_felts: Vec<Felt> = pk_sk_bytes.iter().map(|a| Felt::new(*a as u64)).collect::<Vec<Felt>>();
```

into a `Word`. Maybe we should return `(Word, Vec<Felt>)` whereas `Vec<Felt>` represents the `sk`?

Reply:
You should be able to do it as:

```
    let pk_sk_felts = pk_sk_bytes.iter().map(|a| Felt::new(*a as u64)).collect::<Vec<Felt>>();
    let pk = Word::try_from(&pk_sk_felts[0..4]).unwrap();
    let sk = Word::try_from(&pk_sk_felts[4..]).unwrap();
    (pk, sk)
```

Reply:
Hmm using 

```
pub fn get_new_key_pair_with_advice_map() -> (Word, Word) {
    let keypair: KeyPair = KeyPair::new().unwrap();
    let pk_sk_bytes = keypair.to_bytes();
    let _pk_sk_felts = pk_sk_bytes.iter().map(|a| Felt::new(*a as u64)).collect::<Vec<Felt>>();
        let pk_sk_felts = pk_sk_bytes.iter().map(|a| Felt::new(*a as u64)).collect::<Vec<Felt>>();
    let pk = Word::try_from(&pk_sk_felts[0..4]).unwrap();
    let sk = Word::try_from(&pk_sk_felts[4..]).unwrap();

    (pk, sk)
}
```

results in 

```
thread 'test_faucet_contract_mint_fungible_asset_succeeds' panicked at miden-tx/tests/common/mod.rs:95:48:
called `Result::unwrap()` on an `Err` value: TryFromSliceError(())
```

whereas line 95 

```
    let sk = Word::try_from(&pk_sk_felts[4..]).unwrap();
```

Might it be that the sk part of this is more than what fits into a `Word`?

Reply:
Ahh yess - quite possibly.  I've just taken a look at the key configuration and it looks like theres an expanded public key and a public key commitment so what you have at the moment is perfect. In effect you are returning

```
(pk_commitment, Vec[expanded_public_key, secret_key])
```

Lets leave it as is.  Additionally https://github.com/0xPolygonMiden/crypto/issues/217 doesn't make sense anymore because the expanded pk, sk pair wont fit into words.

Reply:
All good, then I think we are good to go